pandas-stubs/core/series.pyi

```diff
@@ -284,7 +287,7 @@ class Series(IndexOpsMixin[S1], NDFrame):
     @overload
     def __new__(
         cls,
-        data: Scalar | _ListLike | dict[int, Any] | dict[_str, Any] | None,
+        data: Scalar | Iterable | None,
```

Comment:
I don't think `Iterable` is correct here, because we don't (or shouldn't) accept a `set`.

Reply:
Would it be correct to have one Never-overload for `set` and then simply use `data: Any` + specified `dtype`?

Reply:
> Would it be correct to have one Never-overload for `set` and then simply use `data: Any` + specified `dtype`?

I don't know.  I'd rather keep things narrow by just using `Scalar | _ListLike | dict[Hashable, Any] | None` .  Or maybe we can use `Mapping[Hashable, Scalar]`  instead of `dict[Hashable, Any]` .

My feeling is that we should make small increments when we change the stubs so as to not widen things too much.

Reply:
> My feeling is that we should make small increments when we change the stubs so as to not widen things too much.

[sorry, I edited your comment by accident - wanted to quote reply]

I agree about that! But I also feel like we are making ourselves too much work by trying to be as restrictive as possible.

Reply:
> I agree about that! But I also feel like we are making ourselves too much work by trying to be as restrictive as possible.

Possibly.  But I'd rather let the community report issues, and we really haven't had too many issues where we made things too restrictive.  

See what I wrote here 18 months ago about this topic:  https://github.com/pandas-dev/pandas-stubs/blob/main/docs/philosophy.md#narrow-vs-wide-arguments